### PR TITLE
Remove deprecated `exists` API in favor of `fillDescriptions` in `SiPixelQualityESProducer`

### PR DESCRIPTION
**Pull request by ferencek**
#### PR description:

This PR addresses issue https://github.com/cms-sw/cmssw/issues/40077

#### PR validation:

Code compiles and cmsDriver from the original [JIRA](https://its.cern.ch/jira/browse/PDMVMCPROD-62?focusedCommentId=4618412&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-4618412) runs successfully.

#### Backports:

No backports are needed.

### CODE REVIEW

**Comment by mmusich** [comment]
please test

File: CalibTracker/SiPixelESProducers/plugins/SiPixelQualityESProducer.cc [hunk 1]
```diff
@@ -141,4 +142,32 @@ void SiPixelQualityESProducer::setIntervalFor(const edm::eventsetup::EventSetupR
...
+    }
+    {
+      edm::ParameterSet temp;
+      temp.addParameter<std::string>("record", "SiPixelDetVOffRcd");
+      temp.addParameter<std::string>("tag", "");
+      default_ps.push_back(temp);
+    }
+    desc.addVPSet("ListOfRecordToMerge", desc_ps, default_ps);
+  }
+  descriptions.add("siPixelQualityESProducer", desc);
```
Comment (mmusich) [inline CalibTracker/SiPixelESProducers/plugins/SiPixelQualityESProducer.cc:170]: This could be `addWithDefaultLabel`

Comment (tvami) [reply CalibTracker/SiPixelESProducers/plugins/SiPixelQualityESProducer.cc:170]: hi @ferencek given that the whole PR is somewhat "cosmetic" and there is no urgent need for it to get merged, I think it's fine to retrigger tests, it's better to have this done now then to remember it and do it later...

**Review by mmusich — COMMENTED** [review]
In case there's a follow-up to https://github.com/cms-sw/cmssw/pull/40135/files#r1030742953, I have two more "cosmetic" comments.

File: CalibTracker/SiPixelESProducers/plugins/SiPixelQualityESProducer.cc [hunk 2]
```diff
@@ -141,4 +142,32 @@ void SiPixelQualityESProducer::setIntervalFor(const edm::eventsetup::EventSetupR
   oValidity = infinity;
 }
 
+void SiPixelQualityESProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  // siPixelQualityESProducer
```
[the remark this answers is not in the document]
Comment (ferencek) [reply CalibTracker/SiPixelESProducers/plugins/SiPixelQualityESProducer.cc:146]: This was a literal copy-paste from the `edmPythonConfigToCppValidation` executable described in [this TWiki](https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp#Converting_an_Existing_CFI_file). Will remove it.

File: CalibTracker/SiPixelESProducers/plugins/SiPixelQualityESProducer.cc [hunk 3]
```diff
@@ -141,4 +142,32 @@ void SiPixelQualityESProducer::setIntervalFor(const edm::eventsetup::EventSetupR
...
+      temp.addParameter<std::string>("record", "SiPixelQualityFromDbRcd");
+      temp.addParameter<std::string>("tag", "");
+      default_ps.push_back(temp);
+    }
+    {
+      edm::ParameterSet temp;
+      temp.addParameter<std::string>("record", "SiPixelDetVOffRcd");
+      temp.addParameter<std::string>("tag", "");
+      default_ps.push_back(temp);
+    }
```
[the remark this answers is not in the document]
Comment (ferencek) [reply CalibTracker/SiPixelESProducers/plugins/SiPixelQualityESProducer.cc:167]: This was also a literal copy-paste from the `edmPythonConfigToCppValidation` executable with some changes in variable names to make them a bit more descriptive. I am not actually sure what you want to put in the loop. If you are referring to the two `default_ps.push_back(temp);` calls, the `record` parameter differs in the two instances and introducing some `if` statements would seem less elegant that this implementation proposed by `edmPythonConfigToCppValidation`. This I leave as is for now unless I misunderstood what you meant.

[the remark this answers is not in the document]
Comment (mmusich) [reply CalibTracker/SiPixelESProducers/plugins/SiPixelQualityESProducer.cc:166]: > calls, the record parameter differs in the two instances and introducing some if

No need of an `if`, you can just declare an array with record names and loop over it with range-based loop while doing the `push_back`

[the remark this answers is not in the document]
Comment (mmusich) [reply CalibTracker/SiPixelESProducers/plugins/SiPixelQualityESProducer.cc:166]: > calls, the record parameter differs in the two instances and introducing some if

No need of an `if`, you can just declare an array with record names and loop over it with range-based loop while doing the `push_back`

[the remark this answers is not in the document]
Comment (mmusich) [reply CalibTracker/SiPixelESProducers/plugins/SiPixelQualityESProducer.cc:166]: I guess we have different taste in readability. Anyway It does not bother me at all,  it was said upfront it's "cosmetics".

**Comment by sroychow** [comment]
please test

**Comment by tvami** [comment]
+alca
 * tests pass, diffs in MsgLogger only

**Comment by perrotta** [comment]
+1

**Pull request merged**